go/vt/vtgate/engine/fk_cascade.go

```diff
@@ -168,17 +168,30 @@ func (fkc *FkCascade) TryStreamExecute(ctx context.Context, vcursor VCursor, bin
 
 // Inputs implements the Primitive interface.
 func (fkc *FkCascade) Inputs() []Primitive {
-	inputs := []Primitive{fkc.Selection}
-	for _, child := range fkc.Children {
-		inputs = append(inputs, child.Exec)
-	}
-	inputs = append(inputs, fkc.Parent)
-	return inputs
+	return nil
 }
 
 func (fkc *FkCascade) description() PrimitiveDescription {
+	var childrenDesc []PrimitiveDescription
+	for _, child := range fkc.Children {
```

Comment:
don't need to visit children here. this is done outside, from the caller of this method

Reply:
We don't put the `fkc.Children` in the `Inputs()` output so we don't actually visit them from the caller of this method.
The reason we do it this way is because we want to print the BvName and Cols fields too which otherwise aren't printed.

Reply:
We discussed it offline, and we'll change this. cc - @harshit-gangal